### PR TITLE
feat: warn for misordered imports

### PR DESCRIPTION
**Pull request by skeggse**
Not entirely sure what our preferred ordering is - this is just my attempt at putting together some ordering rules that are helpful.

I recommend the use of [ESLint-Formatter](https://github.com/TheSavior/ESLint-Formatter) for sublime users with [`eslint_d`](https://github.com/mantoni/eslint_d.js) (see https://github.com/roadhump/SublimeLinter-contrib-eslint_d/issues/18).

image

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -92,6 +93,16 @@ module.exports = {
     // rely on implicit returns when return booleans or undefined.
     'consistent-return': ['error', { treatUndefinedAsUnspecified: true }],
     'eol-last': ['error', 'always'],
+
+    'import/order': [
+      'warn',
+      {
+        alphabetize: { order: 'asc', caseInsensitive: true },
+        'newlines-between': 'always',
+        pathGroups: [{ pattern: '{,~}/**', group: 'internal' }],
```
Comment (janecakemaster) [inline index.js:102]: niceee

File: index.js [hunk 2]
```diff
@@ -92,6 +93,16 @@ module.exports = {
...
+        groups: ['builtin', 'external', ['internal', 'parent', 'sibling'], 'index'],
```
Comment (janecakemaster) [inline index.js:103]: also niceee

**Comment by janecakemaster** [comment]
my preference is to have the files closer to the one you're inside at the bottom (so `./blahblah` goes AFTER `/utils/blahblah`)

File: index.js (hunk 1 above)
Comment (shils) [inline index.js:102]: What kind of imports does the comma support capturing?

File: index.js (hunk 2 above)
Comment (rfreebern) [reply index.js:103]: For `internal` can we also group `@mixmaxhq` imports together?

File: index.js (hunk 1 above)
Comment (skeggse) [reply index.js:102]: This is bash expansion - this doesn't match a literal comma, but matches both `/**` and `~/**`.

File: index.js (hunk 2 above)
Comment (skeggse) [reply index.js:103]: As a separate section above the `'internal', 'parent', 'sibling'` section?

**Comment by skeggse** [comment]
@rfreebern @janecakemaster please do feel free to propose alternate `groups` specs - this is just a draft!

File: index.js (hunk 2 above)
Comment (skeggse) [reply index.js:103]: Seems like it might be nice to group `/utils` and `@mixmaxhq/*` together, maybe, and in a separate section?

Comment (skeggse) [reply index.js]: Updated!

**Pull request merged**